src/mon/OSDMonitor.cc

```diff
@@ -13095,6 +13095,82 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     wait_for_commit(op, new Monitor::C_Command(mon, op, 0, rs,
 					      get_last_committed() + 1));
     return true;
+  } else if (prefix == "osd pool force-reremove-snap") {
+    string poolstr;
+    cmd_getval(cmdmap, "pool", poolstr);
+    int64_t pool = osdmap.lookup_pg_pool_name(poolstr.c_str());
+    if (pool < 0) {
+      ss << "unrecognized pool '" << poolstr << "'";
+      err = -ENOENT;
+      goto reply_no_propose;
+    }
+
+    const pg_pool_t *p = osdmap.get_pg_pool(pool);
+    pg_pool_t *pp = 0;
+    if (pending_inc.new_pools.count(pool))
+      pp = &pending_inc.new_pools[pool];
+    if (!pp) {
+      pp = &pending_inc.new_pools[pool];
+      *pp = *p;
```

Comment:
This is quirky. It looks we're creating the entry in the container with default ctor (`operator[]`), and then copy-assign from `osdmap`. Could it be simplified or commented?

But in the fist place: why do we need the entry in `new_pools`?

Reply:
Closing as discussed offline.

Reply:
I see, it follows an already established convention. I'm fine with addressing in a follow-up PR.